app.py

```diff
@@ -149,12 +157,15 @@ def table():
 
 @app.route('/delete/contacts/<id>', methods=["POST"])
 @login_required
-def delete(id):
-
-    phonebook.delete(id)
-    flash(f"ID:{id} Deleted")
+def delete_contact(id):
+    try:
+        phonebook.delete(id)
+        flash(f"ID:{id} Deleted")
+        return redirect(url_for('table'))
 
-    return make_response(redirect(url_for('table')))
+    except errors.ForeignKeyViolation:
+        flash('You should delete associated activities first!')
```

Comment:
Users should not be bothered with such details. Find another way for this!

Reply:
We shouldn't warn them about deleteing a contact with activities ?! 
I can delete it instantly, but I thought maybe it's not a good idea!

can you be more specific on that?